lib/bot/giveHelp.js

```diff
@@ -1,28 +1,30 @@
 'use strict';
+const fs = require('fs');
+const mustache = require('mustache');
+const timezone = process.env.TIMEZONE || 'America/New_York';
+const log = require('../../getLogger')('help giver');
 
+const overviewDoc = fs.readFileSync('./documentation/overview.md', { encoding: 'utf-8' });
 
 function attachListener(controller, botName) {
   controller.hears(['^(help|usage)'],['direct_mention','direct_message'], function(bot, message) {
-    var helpMessage =
-
-      '- To create a standup, say `@'+botName+' create standup [time]` in a channel. That\'ll create a standup for every weekday. You can also specify just certain weekdays: `@'+botName+' create standup [time] [days]`, where `days` is a list of space-separate days, such as `M W F` or `T Th`\n'+
-      '- To create a reminder for that standup, say `@'+botName+' reminder [minutes before standup]`\n'+
-      '- To log your standup response, either emoji that reminder or DM me with the following format '+
-      '```#channel-name\ny: what I did yesterday\nT: what I\'m doing today\n'+
-      'b: what\'s in my way\ng: what main thing I hope to accomplish```\n'+
-      '- To remove a standup, say `@'+botName+' remove standup` in the channel where it\'s '+
-      'scheduled\nBy default, notifications are sent with @-here, but you can change that to '+
-      'something else—like a user group or @-channel—by saying `@'+botName+
-      ' audience [notification]`\n'+
-      '- To get a report of past standup responses, DM me with `report` or say `@'+botName+' report` in the '+
-      'channel you\'re curious about. To customize that report,'+
-      ' say any combination of `report [user] [channel] [days]`';
-
+    let fullHelp = '';
     if(process.env.URL) {
-      helpMessage += '\n\nFor more help, see ' + process.env.URL;
+      fullHelp = `Here's some helpful tips.  You can also check out my [full standup-bot documentation](${process.env.URL}).\n\n`;
     }
+    const helpFile = mustache.render(overviewDoc, { bot_name: botName, timezone, full_help: fullHelp });
 
-    bot.reply(message, helpMessage);
+    bot.api.files.upload({
+      title:'standup-bot help',
+      filename: 'standup-bot-help.md',
+      filetype: 'post',
+      content: helpFile,
+      channels: message.channel
+    }, function(err) {
+      if(err) {
+        log.error(err);
```

Comment:
Yeah, probably so.  I only added the callback to make the tests happy because the mock has a `yield` and it threw an exception when there wasn't a callback...  Not the best reason, but there you go.